insane.go

```diff
@@ -144,11 +143,24 @@ type StrictNode struct {
 }
 
 type decoder struct {
-	id        int
-	buf       []byte
-	root      Root
-	nodePool  []*Node
-	nodeCount int
+	id       int
+	buf      []byte
+	root     Root
+	nodePool []Node
+}
+
+func (d *decoder) nodeFromPool() *Node {
```

Comment:
This function is inlinable:
`./insane.go:152:6: can inline (*decoder).nodeFromPool`
so the new function will not reduce performance due to its call